tests/integration/aof-multi-part.tcl

```diff
@@ -1500,10 +1500,19 @@ tags {"external:skip"} {
                 fail "No write load detected."
             }
 
-            # Get the master_repl_offset when trigger bgrewriteaof
-            r config set appendonly no
-            set offset1 [s master_repl_offset]
-            r config set appendonly yes
+            # We obtain the master_repl_offset at the time of bgrewriteaof by pausing
+            # the redis process, sending pipeline commands, and then resuming the process
+            set rd [redis_deferring_client]
+            pause_process [srv 0 pid]
```

Comment:
so now we can add a replica or enable AOF?
you said that previously the only reason it worked was because they were missing

Reply:
yes, AOF is enabled when starting server, now, we can add a replica, your suggestion makes it robust

Reply:
i didn't notice in the diffs that you added AOF.

Reply:
I don't follow you, it is not necessary to add a replica, if aof is enabled, `master_repl_offset` also will be incremented even there is no replica. https://github.com/redis/redis/blob/unstable/src/replication.c#L511
Did I miss something?

Reply:
but you previously said:
> I found master_repl_offset is not changed because there is no replica and AOF is disable

i.e. the only reason the tests passed is because AOF was disabled. and i didn't notice you changed that.

Reply:
ohh, i missed that (and didn't bother to look closely)